### ISSUE TITLE
undefined symbol 

### ISSUE THREAD

**Issue by TonyPeng-2018** [OpenDriveLab/UniAD#85]
Hello,

Full Error:
*****************************************
Traceback (most recent call last):
  File "./tools/train.py", line 19, in <module>
    from mmdet3d.datasets import build_dataset
  File "xxx/workspace/mmdetection3d/mmdet3d/datasets/__init__.py", line 4, in <module>
    from .custom_3d import Custom3DDataset
  File "xxx/workspace/mmdetection3d/mmdet3d/datasets/custom_3d.py", line 10, in <module>
    from ..core.bbox import get_box_type
  File "xxx/workspace/mmdetection3d/mmdet3d/core/__init__.py", line 3, in <module>
    from .bbox import *  # noqa: F401, F403
  File "xxx/workspace/mmdetection3d/mmdet3d/core/bbox/__init__.py", line 5, in <module>
    from .iou_calculators import (AxisAlignedBboxOverlaps3D, BboxOverlaps3D,
  File "/xxx/workspace/mmdetection3d/mmdet3d/core/bbox/iou_calculators/__init__.py", line 2, in <module>
    from .iou3d_calculator import (AxisAlignedBboxOverlaps3D, BboxOverlaps3D,
  File "xxx/workspace/mmdetection3d/mmdet3d/core/bbox/iou_calculators/iou3d_calculator.py", line 6, in <module>
    from ..structures import get_box_type
  File "xxx/workspace/mmdetection3d/mmdet3d/core/bbox/structures/__init__.py", line 2, in <module>
    from .base_box3d import BaseInstance3DBoxes
  File "xxx/workspace/mmdetection3d/mmdet3d/core/bbox/structures/base_box3d.py", line 6, in <module>
    from mmdet3d.ops.iou3d import iou3d_cuda
  File "xxx/workspace/mmdetection3d/mmdet3d/ops/__init__.py", line 20, in <module>
    from .roiaware_pool3d import (RoIAwarePool3d, points_in_boxes_batch,
  File "xxx/workspace/mmdetection3d/mmdet3d/ops/roiaware_pool3d/__init__.py", line 1, in <module>
    from .points_in_boxes import (points_in_boxes_batch, points_in_boxes_cpu,
  File "xxx/workspace/mmdetection3d/mmdet3d/ops/roiaware_pool3d/points_in_boxes.py", line 3, in <module>
    from . import roiaware_pool3d_ext
**ImportError: xxx/workspace/mmdetection3d/mmdet3d/ops/roiaware_pool3d/roiaware_pool3d_ext.cpython-38-x86_64-linux-gnu.so: undefined symbol: _ZNK3c107SymBool10guard_boolEPKcl**

System: Ubuntu 18.04

I followed all steps in INSTALL.md and DATA_PREP.md except:
1. **export PATH=/usr/lib/gcc/x86_64-linux-gnu/5/:$PATH**, the path is not found, so I run **conda install -c omgarcia gcc-6** directly
2. I got an error after I run **cd UniAD pip install -r requirements.txt**, the error is shown below:
**ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
nuscenes-devkit 1.1.10 requires matplotlib<=3.5.2, but you have matplotlib 3.7.2 which is incompatible.
mmdet3d 0.17.1 requires numpy<1.20.0, but you have numpy 1.24.4 which is incompatible.** I ignore it.  

Then I tried command:
**./tools/uniad_dist_train.sh ./projects/configs/stage1_track_map/base_track_map.py N_GPUS**
and met the error. 

Try to fix it:
**I reinstall mmcv-full==1.4.0 mmdet==2.14.0 mmsegmentation==0.14.1
git clone https://github.com/open-mmlab/mmdetection3d.git
cd mmdetection3d
git checkout v0.17.1
pip install scipy==1.7.3 --force-reinstall
pip install scikit-image==0.20.0 --force-reinstall
pip install -v -e .
git clone https://github.com/OpenDriveLab/UniAD.git
cd UniAD
pip install -r requirements.txt**
This issue still exists.

**Comment by TonyPeng-2018** [comment]
I think the problem is caused by gcc. The gcc used for installing and for compiling are not the same version. I will reinstall the mmcv and mmdetection3d again.

**Comment by TonyPeng-2018** [comment]
fixed.

**Comment by tunmx** [comment]
Hello, can you share how you did it, thank you！

**Comment by iyuner** [comment]
Hi! I also met similar error. `envs/uniad/lib/python3.8/site-packages/mmcv/_ext.cpython-38-x86_64-linux-gnu.so: undefined symbol: _ZNK2at10TensorBase8data_ptrIdEEPT_v`
It seems the problem of mmcv installation. I fixed it, by reinstalling mmcv 1.4.0 from the source with the gcc installed in the system.

```
cd mmcv & export MMCV_WITH_OPS=1 && export FORCE_CUDA=1 && pip install -e .
```
Then 
```
pip install mmdet==2.14.0 mmsegmentation==0.14.1
git clone https://github.com/open-mmlab/mmdetection3d.git
cd mmdetection3d
git checkout v0.17.1
pip install scipy==1.7.3 --force-reinstall
pip install scikit-image==0.20.0 --force-reinstall
pip install -v -e .
git clone https://github.com/OpenDriveLab/UniAD.git
cd UniAD
pip install -r requirements.txt
```
Then it works for me.
